CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/block/BlockStateMatcher.java

```diff
@@ -0,0 +1,82 @@
+package crafttweaker.mc1120.block;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import crafttweaker.annotations.ZenRegister;
+import crafttweaker.api.block.*;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.IBlockState;
+import stanhebben.zenscript.annotations.ZenClass;
+import stanhebben.zenscript.annotations.ZenMethod;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@ZenClass("crafttweaker.mc1120.block.BlockStateMatcher")
+@ZenRegister
+public class BlockStateMatcher implements IBlockStateMatcher {
+
+    private final crafttweaker.api.block.IBlockState blockState;
+    private final Map<String, List<String>> allowedProperties;
+
+    public BlockStateMatcher(crafttweaker.api.block.IBlockState blockState, Map<String, List<String>> allowedProperties) {
+        this.blockState = blockState;
+        this.allowedProperties = ImmutableMap.copyOf(allowedProperties);
+    }
+
+    public BlockStateMatcher(crafttweaker.api.block.IBlockState blockState) {
+        this(blockState, ImmutableMap.of());
+    }
+
+    @ZenMethod
+    public static IBlockStateMatcher create(crafttweaker.api.block.IBlockState... blockStates) {
+        if (blockStates == null || blockStates.length == 0) return null;
```

Comment:
Validations should be mentioned if returning null. Also what happens if you put in the bracket handler and it returns null values here so you end up with an array of 1 null.

Reply:
Also don't recall what the style guide is, but generally I avoid same line returns like that.